src/js/libgeo/GeoRender.js

```diff
@@ -40,6 +40,25 @@ function render() {
         }
     }
 
+    function drawgeoarc(el) {
+        if (!el.isshowing || el.visible === false)
+            return;
+
+        var modifs = {};
+        modifs.color = el.color;
+        modifs.alpha = el.alpha;
+        modifs.size = el.size;
+
+        // check if we have filled: true
+        var df = el.filled ? "F" : "D";
+
+        var newargs = [csgeo.csnames[el.args[0]].homog,
+            csgeo.csnames[el.args[1]].homog,
+            csgeo.csnames[el.args[2]].homog
+        ];
+        eval_helper.drawarc(newargs, modifs, df);
+    }
```

Comment:
Here the conversion from argument names to their coordinates is right. This conversion from `el.args` to `newargs` should be enough to make the other similar changes unneccessary.